### DISCUSSION TITLE
I have used Nuxt server components to perform backend job

### DISCUSSION THREAD

**Discussion by ElvisAns** [nuxt/nuxt#24776] [Sharing]
![IHaveAQuestionDavidCraneGIF](https://github.com/nuxt/nuxt/assets/35831811/eea69de7-92c8-4424-b5cb-ad60c81c87e5)

I have used Nuxt server components to perform backend job; i didn't want to build an entire API with just one or 2 routes..
It was painful to pass the backend data back to my nuxt app because the server component is supposed to be self contained or at least take some props from parent but not reverse, i have managed this by using MutationObserver and get the API response from .textContent of the server component. 

⁉️ **is this an anti pattern?**

**Comment by TheAlexLichter** [comment]
From what you describe - yes. What speaks against creating the API routes straight away in the `server` folder instead (+ calling the API via useFetch).

**Reply by ElvisAns** [reply]
Thanks @manniL, i will thoroughly look at the documentation about this.
But the goal is to do all this backend call in server environment not the browser

**Reply by TheAlexLichter** [reply]
@ElvisAns Then your nitro endpoint can *call* another API endpoint 😊

**Reply by ElvisAns** [reply]
Just wanted to thank you @manniL , everything is going well for my project and I am even turning into a Nitro ambassador haha 
See my intro article about it [introducing-nitro-by-unjs-simplifying-api-development-in-nodejs]( https://dev.to/elvisans/introducing-nitro-by-unjs-simplifying-api-development-in-nodejs-4bkh)
